### ISSUE TITLE
[add] Hash only gives wrong result on directories when using CID

### ISSUE THREAD

**Issue by djdv** [ipfs/go-ipfs#4065]
#### Version information:
go-ipfs version: 0.4.11-dev-5ba22a0c7
Repo version: 5
System version: amd64/windows
Golang version: go1.8.3

#### Type:
Bug

#### Severity:
High
#### Description:
When using --cid-version=1 combined with -n on a directory and/or with -w, the wrong hash is returned.
```
>mkdir empty
>ipfs add -r --cid-version=0 empty
added QmUNLLsPACCz1vLxQVkXqqLX5R1X345qqfHbsf67hvA3Nn empty
>ipfs add -r --cid-version=1 empty
added zdj7WbTaiJT1fgatdet9Ei9iDB5hdCxkbVyhyh8YTUnXMiwYi empty
>ipfs add -r -n --cid-version=1 empty
added QmUNLLsPACCz1vLxQVkXqqLX5R1X345qqfHbsf67hvA3Nn empty
```

**Comment by Kubuxu** [comment]
This needs to get fixed. @magik6k do you have time to take a look at it?

**Comment by kevina** [comment]
@Kubuxu I can look into this, I am the one responsible for most of the CidV1 code in the adder.  Is there a reason you want @magik6k to look into it?

**Comment by kevina** [comment]
And this appears to be fixed in 0.4.12 (likely due to my code adding CIDv1 support to the files interface which uses related code).

```
$ ipfs --version
ipfs version 0.4.12-rc2
$ ipfs add -r --cid-version=0 empty
added QmUNLLsPACCz1vLxQVkXqqLX5R1X345qqfHbsf67hvA3Nn empty
 0 B / 4.00 KB [-------------------------------------------------------]   0.00%
$ ipfs add -r --cid-version=1 empty
added zdj7WbTaiJT1fgatdet9Ei9iDB5hdCxkbVyhyh8YTUnXMiwYi empty
 0 B / 4.00 KB [-------------------------------------------------------]   0.00%
$ ipfs add -r -n --cid-version=1 empty
added zdj7WbTaiJT1fgatdet9Ei9iDB5hdCxkbVyhyh8YTUnXMiwYi empty
 0 B / 4.00 KB [-------------------------------------------------------]   0.00%
```

**Comment by Kubuxu** [comment]
Cool.
